api/saml/configuration.py

```diff
@@ -0,0 +1,376 @@
+
+        return value
+
+
+class SAMLMetadataSerializer(SAMLConfigurationSerializer):
+    """Serializes and deserializes values as pickled library's configuration settings"""
+
+    def __init__(self, integration):
+        """Initializes a new instance of SAMLMetadataSerializer class
+
+        :param integration: External integration
+        :type integration: ExternalIntegration
+        """
+        super(SAMLMetadataSerializer, self).__init__(integration)
+
+    def serialize(self, setting_name, value):
+        """Serializes the value as a pickled library's configuration setting
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :param value: Value to be serialized (should be picklable)
+        :type value: Any
+        """
+        if not value:
+            raise ValueError('Value must be non-empty')
+
+        serialized_value = pickle.dumps(value)
+        super(SAMLMetadataSerializer, self).serialize(setting_name, serialized_value)
+
+    def deserialize(self, setting_name):
+        """Deserializes and returns the library's configuration setting
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :return: Any
+        """
+        serialized_value = super(SAMLMetadataSerializer, self).deserialize(setting_name)
+
+        if not serialized_value:
+            raise SAMLConfigurationSerializingError(message=_('Serialized value is empty'))
+
+        deserialized_value = pickle.loads(serialized_value)
+
+        return deserialized_value
+
+
+class SAMLConfigurationError(SAMLError):
+    """Raised in the case of any configuration errors"""
+
+
+class SAMLConfiguration(object):
+    """Contains SP and IdP settings"""
+
+    DEBUG = 'debug'
+    STRICT = 'strict'
+
+    SP_METADATA = 'sp_metadata'
+    SP_XML_METADATA = 'sp_xml_metadata'
+    SP_PRIVATE_KEY = 'sp_private_key'
+
+    IDP_XML_METADATA = 'idp_xml_metadata'
+    IDP_METADATA = 'idp_metadata'
+
+    def __init__(self, configuration_serializer, metadata_serializer):
+        """Initializes a new instance of SAMLConfiguration class
+
+        :param configuration_serializer: SAML configuration serializer
+        :type configuration_serializer: SAMLConfigurationSerializer
+
+        :param metadata_serializer: SAML metadata configuration serializer
+        :type metadata_serializer: SAMLMetadataSerializer
+        """
+        self._configuration_serializer = configuration_serializer
+        self._metadata_serializer = metadata_serializer
+
+        self._debug = None
+        self._strict = None
+
+        self._identity_providers = None
+        self._service_provider = None
+
+    def _load_debug(self):
+        """Returns a debug mode indicator
+
+        :return: Debug mode indicator
+        :rtype: bool
+        """
+        debug = bool(self._configuration_serializer.deserialize(self.DEBUG))
```

Comment:
Here's one of the weird moments I mentioned earlier. We're "deserializing" a value by pulling a string out of the database (not what I would call "deserializing") and then converting it from a string to a bool (what I _would_ call "deserializing"). 

I don't want to make a big deal out of this because it's not the most complicated part of this branch, but in combination with the pickle thing (which is in play in metadata_serializer but not configuration_serializer) it makes this file confusing to read.